sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -818,7 +820,7 @@ private async Task<AmqpSession> CreateAndOpenSessionAsync(AmqpConnection connect
                 {
                     Role = false,
                     InitialDeliveryCount = 0,
-                    Source = new Source { Address = Guid.NewGuid().ToString() },
+                    Source = new Source { Address = identifier },
```

Comment:
I'm not sure we will leverage this feature yet for Durable Terminus, but my understanding is that link recovery is based on link name. It looks like the LinkName is set on L831 and includes the identifier, but also additional pieces. Should the linkanem just be the identifier? I suppose this could result in uniqueness issues. We may want to confirm with service folks here. /cc @jsquire

Reply:
We've been using the Address in Event Hubs and in Service Bus for the other languages; I don't **_think_** this is the link name they were talking about for durable terminus, but if it is - yeah, we'll want to make this artificial until we decide to enable the resume feature.

Reply:
The service team has said that the link recovery is based on the LinkName + ConnectionId.

Right now the Identifier influences a piece of the LinkName. Is the plan that if we need to support link recovery through the LinkIdentifier, that we would add a different setting to specify the link recovery link? Or would we just take durable terminus being enabled to mean that Identifier should be used as the LinkName + ConnectionId?

Reply:
My vote would be the latter.   I don't see a reason to have multiple identifiers for clients with slightly different purposes.    Personally, I'd like to see the ConnectionId stay random today and the LinkName get set to the Identifier.    That way, if we light up terminus recovery, we only have to pipe the `ServiceBusClient.Identifier` into the connection id.

Reply:
Makes sense to me, but does that mean the Identifier behavior would differ between SB and EH, and .NET SB and other language SB?

Reply:
@JoshLove-msft @jsquire does it make more sense for the sender link to be associated with the sender identifier or the general service bus client identifier? As it's written these are independent

Reply:
We should use the sender/receiver identifiers with the links.   Eventually, we'll take the `ServiceBusClient` identifier and associate it with the connection... but, for now, we want to keep the connection's name generated so that we don't unintentionally enable a feature.  :)

Reply:
> Makes sense to me, but does that mean the Identifier behavior would differ between SB and EH, and .NET SB and other language SB?

It shouldn't.   Event Hubs uses the identifier as the Source/Target address currently.   That was also in the issue for the other languages to do in SB.    

The Link Name pattern will need to change a bit to make it predictable for durable terminus recovery - so we should probably drop the session/link parts.   That will deviate from EH, but should have the same pattern for SB in other languages after they do the durable terminus stuff.   That said, I've no objection to leaving the link name as-is for now and considering changes as part of the durable terminus story later.    Your call, @JoshLove-msft, I support whichever direction you decide is best.

Reply:
As long as there are no concerns from the service side for dropping the session/link part from the session name, I would vote to do it now just in case we end up needing this for durable terminus.

Reply:
Actually, I think we should just leave as-is for now. We can always modify the behavior based on whether durable terminus receive mode is being used. And there is a good chance we will never enable this part of the durable terminus story either.

Reply:
Sounds good. Switched it back.